src/main/java/seedu/address/model/internship/TaskList.java

```diff
@@ -0,0 +1,131 @@
+package seedu.address.model.internship;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks in the internship book.
+ */
+public class TaskList {
+    private final ArrayList<Task> taskList;
+
+    /**
+     * Constructs a TaskList object.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a TaskList object from an existing list of tasks.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Adds tasks from a comma-separated string into the task list. Only for testing purposes using TypicalInternships.
+     *
+     * @param tasksString the string containing comma-separated tasks
+     */
+    public TaskList(String tasksString) {
+        if (tasksString.equals("") || tasksString.equals(" ")) {
+            this.taskList = new ArrayList<>();
+        } else {
+            ArrayList<Task> taskList = new ArrayList<>();
+            String[] tasksArray = tasksString.split(", ");
+            for (String taskDescription : tasksArray) {
+                taskList.add(new Task(taskDescription.trim()));
+            }
+            this.taskList = taskList;
+        }
+    }
+
+    /**
+     * Returns the taskList with {@code ArrayList<Task>} type. Primarily for JSON purposes.
+     */
+    public ArrayList<Task> getArrayListTaskList() {
+        return taskList;
+    }
+
+    public void addTask(Task task) {
+        taskList.add(task);
+    }
+
+    public Task getTask(int index) {
+        // Index is 1-based
```

Comment:
This comment is misleading, you actually pass in 0-based indexes into this function